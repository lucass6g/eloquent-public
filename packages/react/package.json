{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
