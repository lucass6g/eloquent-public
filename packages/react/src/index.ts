export {Switch, Case, Default} from './components/conditional-switch'