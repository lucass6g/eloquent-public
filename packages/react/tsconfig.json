{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
