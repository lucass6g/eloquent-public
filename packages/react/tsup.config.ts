import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})