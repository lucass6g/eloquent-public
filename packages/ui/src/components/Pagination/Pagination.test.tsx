import { render } from "@testing-library/react";
import { Pagination } from "./index";
import { describe, expect, it, vi } from "vitest";

describe("Pagination", () => {
  describe("should match the snapshot", () => {
    it("default", () => {
      const component = render(
        <Pagination.Root>
          <Pagination.Content>
            <Pagination.Item>
              <Pagination.FirstPage disabled={true} />
            </Pagination.Item>
            <Pagination.Item>
              <Pagination.Previous />
            </Pagination.Item>
            <Pagination.Item>
              <Pagination.Button isActive={true}>1</Pagination.Button>
            </Pagination.Item>
            <Pagination.Item>
              <Pagination.Button>2</Pagination.Button>
            </Pagination.Item>
            <Pagination.Item>
              <Pagination.Button>3</Pagination.Button>
            </Pagination.Item>
            <Pagination.Item>
              <Pagination.Next />
            </Pagination.Item>
            <Pagination.Item>
              <Pagination.LastPage onClick={() => console.log("last")} />
            </Pagination.Item>
          </Pagination.Content>
        </Pagination.Root>
      );

      expect(component.baseElement).toMatchSnapshot();
    });
    it("label", () => {
      const component = render(
        <Pagination.Root>
          <Pagination.Label current={1} total={10} />
        </Pagination.Root>
      );

      expect(component.baseElement).toMatchSnapshot();
    });
    it("form", () => {
      const component = render(
        <Pagination.Form
          current={1}
          total={10}
          onChange={vi.fn()}
          onSubmit={vi.fn()}
        />
      );

      expect(component.baseElement).toMatchSnapshot();
    });
  });
});
