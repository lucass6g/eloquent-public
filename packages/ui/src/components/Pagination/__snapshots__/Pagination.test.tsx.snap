// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > should match the snapshot > default 1`] = `
<body>
  <div>
    <nav
      aria-label="pagination"
      role="navigation"
    >
      <ul
        class="flex flex-row items-center gap-1"
      >
        <li>
          <button
            aria-label="Primeira página"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm antialiased leading-4 p-0 border border-border h-10 w-10 first-child:stroke-primary-foreground gap-1 bg-neutral-100 disabled:bg-neutral-100 shadow-none disabled:text-neutral-400 text-neutral-800 hover:bg-neutral-200 font-normal data-[active=true]:bg-neutral-200 data-[active=true]:font-semibold"
            data-active="false"
            disabled=""
          >
            <svg
              class="lucide lucide-chevrons-left h-4 w-4 stroke-[1.5px]"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m11 17-5-5 5-5"
              />
              <path
                d="m18 17-5-5 5-5"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-label="Página anterior"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm antialiased leading-4 p-0 border border-border h-10 w-10 first-child:stroke-primary-foreground gap-1 bg-neutral-100 disabled:bg-neutral-100 shadow-none disabled:text-neutral-400 text-neutral-800 hover:bg-neutral-200 font-normal data-[active=true]:bg-neutral-200 data-[active=true]:font-semibold"
            data-active="false"
          >
            <svg
              class="lucide lucide-chevron-left h-4 w-4 stroke-[1.5px]"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m15 18-6-6 6-6"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm antialiased leading-4 p-0 border border-border h-10 w-10 first-child:stroke-primary-foreground gap-1 bg-neutral-100 disabled:bg-neutral-100 shadow-none disabled:text-neutral-400 text-neutral-800 hover:bg-neutral-200 font-normal data-[active=true]:bg-neutral-200 data-[active=true]:font-semibold"
            data-active="true"
          >
            1
          </button>
        </li>
        <li>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm antialiased leading-4 p-0 border border-border h-10 w-10 first-child:stroke-primary-foreground gap-1 bg-neutral-100 disabled:bg-neutral-100 shadow-none disabled:text-neutral-400 text-neutral-800 hover:bg-neutral-200 font-normal data-[active=true]:bg-neutral-200 data-[active=true]:font-semibold"
            data-active="false"
          >
            2
          </button>
        </li>
        <li>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm antialiased leading-4 p-0 border border-border h-10 w-10 first-child:stroke-primary-foreground gap-1 bg-neutral-100 disabled:bg-neutral-100 shadow-none disabled:text-neutral-400 text-neutral-800 hover:bg-neutral-200 font-normal data-[active=true]:bg-neutral-200 data-[active=true]:font-semibold"
            data-active="false"
          >
            3
          </button>
        </li>
        <li>
          <button
            aria-label="Próxima página"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm antialiased leading-4 p-0 border border-border h-10 w-10 first-child:stroke-primary-foreground gap-1 bg-neutral-100 disabled:bg-neutral-100 shadow-none disabled:text-neutral-400 text-neutral-800 hover:bg-neutral-200 font-normal data-[active=true]:bg-neutral-200 data-[active=true]:font-semibold"
            data-active="false"
          >
            <svg
              class="lucide lucide-chevron-right h-4 w-4 stroke-[1.5px]"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-label="Última página"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm antialiased leading-4 p-0 border border-border h-10 w-10 first-child:stroke-primary-foreground gap-1 bg-neutral-100 disabled:bg-neutral-100 shadow-none disabled:text-neutral-400 text-neutral-800 hover:bg-neutral-200 font-normal data-[active=true]:bg-neutral-200 data-[active=true]:font-semibold"
            data-active="false"
          >
            <svg
              class="lucide lucide-chevrons-right h-4 w-4 stroke-[1.5px]"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 17 5-5-5-5"
              />
              <path
                d="m13 17 5-5-5-5"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>
</body>
`;

exports[`Pagination > should match the snapshot > form 1`] = `
<body>
  <div>
    <form
      aria-label="pagination"
      class="border border-neutral-300 rounded-full pl-4 flex items-center h-12 bg-neutral-100 w-max"
      role="form"
    >
      <span
        class="text-accent-foreground font-normal text-sm"
      >
        Ir para página
      </span>
      <input
        class="flex rounded-[6px] border border-neutral-300 bg-transparent py-1 transition-colors disabled:cursor-not-allowed disabled:opacity-50 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-neutral-400 file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-neutral-500 w-10 h-10 text-center px-0 mx-2 font-medium text-sm text-neutral-800"
        inputmode="numeric"
        max="10"
        min="1"
        type="text"
        value="1"
      />
      <span
        class="text-accent-foreground font-normal text-sm"
      >
        de 
        10
      </span>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:cursor-not-allowed text-sm antialiased leading-4 p-0 border-border h-10 w-10 first-child:stroke-primary-foreground gap-1 bg-neutral-100 disabled:bg-neutral-100 shadow-none disabled:text-neutral-400 text-neutral-800 hover:bg-neutral-200 font-normal data-[active=true]:bg-neutral-200 data-[active=true]:font-semibold border-0 ml-1"
        data-active="false"
        type="submit"
      >
        <svg
          class="lucide lucide-chevron-right h-4 w-4 stroke-[1.5px]"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m9 18 6-6-6-6"
          />
        </svg>
      </button>
    </form>
  </div>
</body>
`;

exports[`Pagination > should match the snapshot > label 1`] = `
<body>
  <div>
    <nav
      aria-label="pagination"
      role="navigation"
    >
      <span
        class="text-accent-foreground font-normal text-sm"
      >
        Página 
        1
         de 
        10
      </span>
    </nav>
  </div>
</body>
`;
